File: 2486/2496.py
class Solution(object):
    def appendCharacters(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: int
        """
        sPos, tPos = 0, 0
        while sPos < len(s) and tPos < len(t):
            if s[sPos] == t[tPos]:
                tPos += 1
            sPos += 1

        return len(t) - tPos
        

File: 2486/2496.java
class Solution {
    public int appendCharacters(String s, String t) {
        int sPos = 0, tPos = 0;
        while(sPos<s.length() && tPos<t.length()){
            if(s.charAt(sPos) == t.charAt(tPos)){
                tPos++;
            }
            sPos++;
        }

        return t.length() - tPos;
    }
}